platform/o.n.bootstrap/src/org/netbeans/ModuleManager.java

```diff
@@ -1681,7 +1681,9 @@ public boolean hasToEnableCompatModules(Set<Module> modules) throws IllegalArgum
     
     private void maybeAddToEnableList(Set<Module> willEnable, Set<Module> mightEnable, Module m, boolean okToFail) {
         if (! missingDependencies(m).isEmpty()) {
-            assert okToFail : "Module " + m + " had unexpected problems: " + missingDependencies(m) + " (willEnable: " + willEnable + " mightEnable: " + mightEnable + ")";
+            if (!okToFail) {
```

Comment:
This seems to change the behaviour also for non-fragment modules (`assert` -> `log`). Was that intended ?

Reply:
This was introduced  by the PR #847 and has changed behaviour of module loading from NetBeans 9.0 to 10. From my point of view it should be an exception or a log message, throwing an assertion error is not the right solution here.